spi_nand_flash/diskio/diskio_nand.c

```diff
@@ -85,17 +85,23 @@ DRESULT ff_nand_ioctl(BYTE pdrv, BYTE cmd, void *buff)
         ESP_GOTO_ON_ERROR(spi_nand_flash_sync(dev), fail, TAG, "sync failed");
         break;
     case GET_SECTOR_COUNT: {
-        uint16_t num_sectors;
+        uint32_t num_sectors;
         ESP_GOTO_ON_ERROR(spi_nand_flash_get_capacity(dev, &num_sectors),
                           fail, TAG, "get_capacity failed");
         *((DWORD *)buff) = num_sectors;
         ESP_LOGV(TAG, "capacity=%"PRIu32"", *((DWORD *) buff));
         break;
     }
     case GET_SECTOR_SIZE: {
-        uint16_t capacity;
+        uint32_t capacity;
         ESP_GOTO_ON_ERROR(spi_nand_flash_get_sector_size(dev, &capacity),
                           fail, TAG, "get_sector_size failed");
+
+        if (capacity > 0xFFFF) {
+            ESP_LOGE(TAG, "sector size too large");
+            return RES_PARERR;
```

Comment:
@igrr 
Do you think this is the appropriate error to return when overflowing?

https://github.com/espressif/esp-idf/blob/23e4823f17a8349b5e03536ff7653e3e584c9351/components/fatfs/src/diskio.h#L18-L24

Reply:
It seems fine, it should prevent FATFS from initializing further.